app/admin/filter.pipe.ts

```diff
@@ -0,0 +1,31 @@
+﻿import { Injectable, Pipe, PipeTransform } from '@angular/core';
+import { User } from '../shared/user/user.model';
+
+@Pipe({
+    name: 'usersFilter'
+})
+
+@Injectable()
+
+export class UsersFilter implements PipeTransform {
+    transform(users: User[], query: string, criterium: string): User[] {
+        if (query !== undefined && users !== null) {
+            // if (criterium === 'First Name') {
+            //     return users.filter(user =>
+            //         (user.firstName.toLowerCase().indexOf(query.toLowerCase()) !== -1));
+            // } else if (criterium === 'Last Name' || criterium === undefined) {
+            //     return users.filter(user =>
+            //         (user.lastName.toLowerCase().indexOf(query.toLowerCase()) !== -1));
+            // } else
```

Comment:
Put some checks around it to check if the criteriums are not null, so we can uncomment the code